protocols/mdns/src/behaviour/iface.rs

```diff
@@ -69,7 +68,11 @@ pub struct InterfaceState {
     ttl: Duration,
 }
 
-impl InterfaceState {
+impl<U, T> InterfaceState<U, T>
+where
+    U: AsyncSocket,
+    T: TimerBuilder + futures::Stream,
```

Comment:
I am surprised that we can say `futures::Stream` here without specifying the `Item` type?

Reply:
In this case, the `Stream::Item` is not used, so it is not required to check it at compile time.